crates/configuration/src/global_settings.rs

```diff
@@ -0,0 +1,129 @@
+use crate::shared::types::{Duration, IpAddress, MultiAddress};
+
+/// Global settings applied to an entire network.
+#[derive(Debug, Clone, PartialEq)]
+pub struct GlobalSettings {
+    /// Whether we should spawn a dedicated bootnode for each chain.
+    /// TODO: commented now until we decide how we want to use this option
+    // spawn_bootnode: bool,
+
+    /// External bootnode address.
+    bootnodes_addresses: Vec<MultiAddress>,
+
+    /// Global spawn timeout in seconds.
+    network_spawn_timeout: Duration,
+
+    /// Individual node spawn timeout.
+    node_spawn_timeout: Duration,
+
+    /// Local IP used to expose local services (including RPC, metrics and monitoring).
+    local_ip: Option<IpAddress>,
+}
+
+impl GlobalSettings {
+    pub fn bootnodes_addresses(&self) -> Vec<&MultiAddress> {
+        self.bootnodes_addresses.iter().collect()
+    }
+
+    pub fn network_spawn_timeout(&self) -> Duration {
+        self.network_spawn_timeout
+    }
+
+    pub fn node_spawn_timeout(&self) -> Duration {
+        self.node_spawn_timeout
+    }
+
+    pub fn local_ip(&self) -> Option<&IpAddress> {
+        self.local_ip.as_ref()
+    }
+}
+
+#[derive(Debug)]
+pub struct GlobalSettingsBuilder {
+    config: GlobalSettings,
```

Comment:
Do we need the inner `config` here? what's the benefit to use it?

Reply:
Here, it forces us to initialize the config through a builder, the only way to construct the GlobalSettings is through the builder ```.build()``` method so we can control how we construct the settings and add validation check in the ```.build()``` to ensure that when we have a GlobalSettings in the code, it's fully validated (for bootnodes addresses and local_ip)